ucam_webauth/flask_glue.py

```diff
@@ -475,9 +476,21 @@ def _check_url(self, wls_response_url):
                          response_part, request.args["WLS-Response"])
             return None
 
+        if response.ver == 1:
+            # protocol version 1 strips everything from ? from the url
+            # when redirecting; clicking the Cancel button is observed
+            # to cause a version-1 response with this behaviour
+            start = wls_response_url.find("?")
```

Comment:
Here, `start` is overwritte, but it's used again on line 479/492.

Since we then compare `wls_response_url[:this_new_start]` and `actual_url[:this_new_start]`, the test passes most of the time; sometimes when it shouldn't.

Further, if wls_response_url does not contain a '?', start becomes -1, and the test fails when it shouldn't.